### DISCUSSION TITLE
Spellcheck line alert persists after corrections

### DISCUSSION THREAD

**Discussion by gaspar-d** [LazyVim/LazyVim#2453] [Q&A] [answered]
Hi people

I only used Vim vanilla for motions but have now started to give it a try to Lazyvim, loving it but with some small caveats.

How do I uncommon words to spellcheck
or
How to deactivate those alerts

<img width="665" alt="image" src="https://github.com/LazyVim/LazyVim/assets/52392593/c70ede57-ca36-4768-a3ce-c5e738867bff">

1. The spellcheck is showing me this, and at first when I 'zg' it says it don't have a spellfile, so I create one at Lazyvim root.

2. So I added the word and the alert keep showing 

3. Tried to re-spell it, save, close and open back, leader + ca or cf no results

**Comment by FlippingBinary** [comment, marked as answer]
I know this discussion is quite old and you've probably figured out a solution by now, but I just decided to look into getting spellchecking in my environment and stumbled on this discussion while trying to figure it out. Your screenshot shows spelling suggestions at the end of the line as if it's coming from a language server in a way that my environment doesn't show. Maybe you were using [`ltex-ls`](https://github.com/valentjn/ltex-ls) or something. I haven't tried using that yet, but I did discover that LazyVim just uses NeoVim's built-in spelling features. The default keymaps seem to work well (at least the ones I tried).

- Go to the next spelling error: `]s`
- Go to the previous spelling error: `[s`
- Mark word as good: `zg`
- Show suggestions: `z=`

Words are added to a file in the LazyVim configuration directory (the one you can access when you press `c` on the dashboard), and the file is created if it doesn't already exist. The `.add` file (like `en.utf-8.add`) is human-readable and the one with the extra `.spl` suffix is a compiled version of it.

I hope this helps someone else who may stumble on this discussion in the future.

**Reply by gubler** [reply]
Thank you, this was very helpful :)
